### ISSUE TITLE
feature request/idea: callahead plugin to verify a RCPT address on a remote server

### ISSUE THREAD

**Issue by jahlives** [haraka/Haraka#1585]
It would be very helpful if haraka would be able to verify a remote RCPT address. Is there any plugin around that could perform this task? If not, would it be something that could be implemented in the core of haraka?
Background: I wrote a plugin that allows relaying for users from a list of domains defined in mysql. But as I only can verify the domain, I'm missing the ability to verify the existance of a user from such domain, prior to accept the RCPT to address. Unfortunately it's not possible to have all those RCPT to in the database as the customers could create new users without inform me to update the users list in mysql.

**Comment by darkpixel** [comment]
There isn't a specific plugin for verifying based on the reply to RCPT TO from a remote server.

You might want to take a look at queue/smtp_proxy (https://haraka.github.io/manual/plugins/queue/smtp_proxy.html).  It relies on responses from the remote server for mail delivery.  You could probably hack up your own copy to read some data from MySQL.

**Comment by msimerson** [comment]
The rcpt_to.ldap plugin verifies remote recipients (via LDAP). The rcpt_to.vpopmail plugin validates remote recipients (via tcp to vpopmaild). The queue/smtp_* plugins do not validate the recipient but instead pass it along and let the remote smtp accept or reject it, and then pass that back.

Call ahead plugins were an okay idea 10 years ago, but they are no longer recommended (sorry, I'm too lazy to look it up in the IETF best practices documents). You can write such a plugin if you like, but I don't recommend holding your breath waiting for the Haraka development team to write one.

**Comment by jahlives** [comment]
Hope it's okay to write here again. First thanks for your replies to a newbie in haraka :-) 
I found [postomaat] (https://github.com/gryphius/postomaat) which has a callahead plugin. After some config effort it works as a callahead for my haraka installation. I just had to write a small client plugin for haraka to query the postomaat daemon.
Hope it helps someone who is looking for this feature (although not recommended practice nowadays :-) )
